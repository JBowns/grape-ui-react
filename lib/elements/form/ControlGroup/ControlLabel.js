'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ControlLabel = undefined;

var _templateObject = _taggedTemplateLiteral(['', ''], ['', '']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _textStyles = require('../../../elements/typography/textStyles');

var _decorators = require('../../../decorators');

var _componentHelpers = require('../../../utils/componentHelpers');

var _globalStyles = require('../../../global-styles');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var _getGlobalStyles = (0, _globalStyles.getGlobalStyles)(),
    gridSchema = _getGlobalStyles.grid;

var padding = gridSchema.gutter;

var ControlLabel = exports.ControlLabel = function ControlLabel(props) {
  var overrides = Object.assign({}, props, {
    sm: true
  });
  var controlLabelStyle = '\n    position: absolute;\n    padding: 0 ' + padding + ';\n    background-color: white;\n    top: ' + padding + ';\n    left: ' + padding + ';\n  ';
  var ProtoControlLabel = (0, _decorators.withBaseStyles)(_styledComponents2.default.label(_templateObject, controlLabelStyle), overrides);
  return _react2.default.createElement(ProtoControlLabel, (0, _componentHelpers.passThrough)(ControlLabel, props));
};

ControlLabel.propTypes = {
  color: _propTypes2.default.string,
  fontFamily: _propTypes2.default.string,
  fontWeight: _propTypes2.default.string,
  kerning: _propTypes2.default.string,
  lg: _propTypes2.default.bool,
  sm: _propTypes2.default.bool,
  textAlign: _propTypes2.default.string,
  textDecoration: _propTypes2.default.string
};

ControlLabel.defaultProps = {
  color: _textStyles.defaultTextStylesBase.color,
  fontFamily: _textStyles.defaultTextStylesBase.fontFamily,
  fontWeight: _textStyles.defaultTextStylesBase.fontWeight,
  kerning: _textStyles.defaultTextStylesBase.kerning,
  lg: _textStyles.defaultTextStylesBase.lg,
  sm: _textStyles.defaultTextStylesBase.sm,
  textAlign: _textStyles.defaultTextStylesBase.textAlign,
  textDecoration: _textStyles.defaultTextStylesBase.textDecoration
};