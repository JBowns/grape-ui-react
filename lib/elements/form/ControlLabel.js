'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ControlLabel = undefined;

var _templateObject = _taggedTemplateLiteral([''], ['']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _textStyles = require('../typography/textStyles');

var _decorators = require('../../decorators');

var _componentHelpers = require('../../utils/componentHelpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var ControlLabel = exports.ControlLabel = function ControlLabel(props) {
  var overrides = Object.assign({}, props, {
    sm: true
  });
  var ProtoControlLabel = (0, _decorators.withBaseStyles)(_styledComponents2.default.label(_templateObject), overrides);
  return _react2.default.createElement(ProtoControlLabel, (0, _componentHelpers.passThrough)(ControlLabel, props));
};

ControlLabel.propTypes = {
  color: _propTypes2.default.string,
  fontFamily: _propTypes2.default.string,
  fontWeight: _propTypes2.default.string,
  kerning: _propTypes2.default.string,
  lg: _propTypes2.default.bool,
  sm: _propTypes2.default.bool,
  textAlign: _propTypes2.default.string,
  textDecoration: _propTypes2.default.string
};

ControlLabel.defaultProps = {
  color: _textStyles.defaultTextStylesBase.color,
  fontFamily: _textStyles.defaultTextStylesBase.fontFamily,
  fontWeight: _textStyles.defaultTextStylesBase.fontWeight,
  kerning: _textStyles.defaultTextStylesBase.kerning,
  lg: _textStyles.defaultTextStylesBase.lg,
  sm: _textStyles.defaultTextStylesBase.sm,
  textAlign: _textStyles.defaultTextStylesBase.textAlign,
  textDecoration: _textStyles.defaultTextStylesBase.textDecoration
};