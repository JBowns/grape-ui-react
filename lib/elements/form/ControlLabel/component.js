"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ControlLabelComponent = undefined;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _componentHelpers = require("../../../utils/componentHelpers");

var _styledHelpers = require("../../../utils/styledHelpers");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var propsToTrim = _objectSpread({}, _styledHelpers.typography.propTypes, {
  activeColor: '',
  disabled: false,
  isRelative: false,
  validationError: ''
});

var ControlLabelComponent = exports.ControlLabelComponent = function ControlLabelComponent(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, ["children"]);

  return (// eslint-disable-next-line jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for
    _react2["default"].createElement("label", (0, _componentHelpers.removeSomeProps)(props, Object.keys(propsToTrim)), children)
  );
};

ControlLabelComponent.propTypes = {
  children: _propTypes2["default"].any
};
ControlLabelComponent.defaultProps = {
  children: null
};