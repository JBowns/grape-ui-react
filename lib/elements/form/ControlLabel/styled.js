'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ControlLabel = undefined;

var _templateObject = _taggedTemplateLiteral(['\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  left: ', ';\n  padding: 0 ', ';\n  ', ';\n  ', '\n'], ['\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  left: ', ';\n  padding: 0 ', ';\n  ', ';\n  ', '\n']);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _globalStyles = require('../../../global-styles');

var _utils = require('../../../elements/typography/utils');

var _styledSystem = require('styled-system');

var _styledHelpers = require('../../../utils/styledHelpers');

var _component = require('./component');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var _getGlobalStyles = (0, _globalStyles.getGlobalStyles)(),
    gridSchema = _getGlobalStyles.grid;

var padding = (0, _utils.getScaledSize)(gridSchema.gutter, 0.5);
var colorLabel = function colorLabel(props) {
  return (0, _styledHelpers.colorCore)(Object.assign({}, props, {
    color: !props.validationError ? props.color : 'brandDanger'
  }));
};
var fontSizeLabel = function fontSizeLabel(props) {
  return (0, _styledHelpers.fontSizeCore)(Object.assign({}, props, { sm: true }));
};
var positionStyle = function positionStyle(props) {
  return 'position: ' + (props.isRelative ? 'relative' : 'absolute') + ';';
};
var top = function top(props) {
  return props.isRelative ? '' : 'top: -' + padding + ';';
};
var ControlLabel = exports.ControlLabel = (0, _styledComponents2.default)(_component.ControlLabelComponent)(_templateObject, colorLabel, _styledHelpers.fontFamilyCore, fontSizeLabel, _styledHelpers.fontStyleCore, _styledSystem.fontWeight, _styledHelpers.letterSpacingCore, _styledHelpers.lineHeightCore, _styledHelpers.fontStyleCore, _styledHelpers.textAlignCore, _styledHelpers.textDecorationCore, padding, padding, top, positionStyle);

ControlLabel.propTypes = Object.assign({
  bg: _propTypes2.default.string,
  disabled: _propTypes2.default.bool,
  isRelative: _propTypes2.default.bool
}, _styledHelpers.typography.propTypes, {
  validationError: _propTypes2.default.string
});

ControlLabel.defaultProps = {
  bg: _styledHelpers.defaultControlStyles.bg,
  disabled: false,
  isRelative: false,
  validationError: ''
};