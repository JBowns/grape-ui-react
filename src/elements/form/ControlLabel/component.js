import React from 'react';
import PropTypes from 'prop-types';
import { passThrough, removeSomeProps } from 'src/utils/componentHelpers';
import { defaultControlStylesBase } from 'src/elements/form/ControlGroup/baseControlStyle';
import { typography } from 'src/elements/typography/textStyles';

export const ControlLabelComponent = ({ children, ...props }) => (
  // eslint-disable-next-line jsx-a11y/label-has-for
  <label {...passThrough(ControlLabelComponent, removeSomeProps(props, ['isRelative']))}>
    {children}
  </label>);
ControlLabelComponent.propTypes = {
  activeColor: PropTypes.string,
  bgColor: PropTypes.string,
  disabled: PropTypes.bool,
  ...typography.propTypes,
  validationError: PropTypes.string,
};

ControlLabelComponent.defaultProps = {
  activeColor: defaultControlStylesBase.activeColor,
  bgColor: defaultControlStylesBase.bgColor,
  disabled: false,
  validationError: '',
};
