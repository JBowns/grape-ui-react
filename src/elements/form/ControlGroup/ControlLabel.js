import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { defaultTextStylesBase } from 'src/elements/typography/textStyles';
import { withBaseStyles } from 'src/decorators';
import { passThrough } from 'src/utils/componentHelpers';
import { getGlobalStyles } from 'src/global-styles';
import { getScaledSize } from 'src/elements/typography/utils';
import { defaultControlStylesBase } from './baseControlStyle';

const { grid: gridSchema } = getGlobalStyles();

const padding = getScaledSize(gridSchema.gutter, 0.5);

export const ControlLabel = props => {
  const { bgColor, color, validationError } = props;
  const overrides = {
    ...props,
    color: !validationError ? color : 'brandDanger',
    sm: true,
  };
  const controlLabelStyle = `
    background-color: ${bgColor};
    left: ${padding};
    padding: 0 ${padding};
    position: absolute;
    top: -${padding};
  `;
  const ProtoControlLabel = withBaseStyles(styled.label`${controlLabelStyle}`, overrides);
  return <ProtoControlLabel {...passThrough(ControlLabel, props)} />;
};

ControlLabel.propTypes = {
  bgColor: PropTypes.string,
  color: PropTypes.string,
  fontFamily: PropTypes.string,
  fontWeight: PropTypes.string,
  kerning: PropTypes.string,
  lg: PropTypes.bool,
  sm: PropTypes.bool,
  textAlign: PropTypes.string,
  textDecoration: PropTypes.string,
  validationError: PropTypes.string,
};

ControlLabel.defaultProps = {
  bgColor: defaultControlStylesBase.bgColor,
  color: defaultTextStylesBase.color,
  fontFamily: defaultTextStylesBase.fontFamily,
  fontWeight: defaultTextStylesBase.fontWeight,
  kerning: defaultTextStylesBase.kerning,
  lg: defaultTextStylesBase.lg,
  sm: defaultTextStylesBase.sm,
  textAlign: defaultTextStylesBase.textAlign,
  textDecoration: defaultTextStylesBase.textDecoration,
  validationError: '',
};
