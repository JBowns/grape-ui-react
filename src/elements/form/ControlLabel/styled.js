import PropTypes from 'prop-types';
import styled from 'styled-components';
import { getGlobalStyles } from 'src/global-styles';
import { getScaledSize } from 'src/elements/typography/utils';
import { resolveColor } from 'src/utils/componentHelpers';
import { defaultControlStylesBase } from 'src/elements/form/ControlGroup/baseControlStyle';
import {
  getFontFamily,
  getFontSize,
  getFontWeight,
  getLetterSpacing,
  getLineHeight,
  getFontStyle,
  getColor,
  getTextAlign,
  getTextDecoration,
  typography,
} from 'src/elements/typography/textStyles';
import { ControlLabelComponent } from './component';

const { grid: gridSchema } = getGlobalStyles();

const padding = getScaledSize(gridSchema.gutter, 0.5);
const bgColor = props => `background-color: ${resolveColor(props.bgColor)};`;
const colorForLabel = props => `${getColor({
  ...props,
  color: !props.validationError ? props.color : 'brandDanger',
})};`;
const fontSizeForLabel = props => `${getFontSize({ ...props, sm: true })};`;
const positionStyle = props => `position: ${props.isRelative ? 'relative' : 'absolute'};`;
const topStyle = props => (props.isRelative ? '' : `top: -${padding};`);
export const ControlLabel = styled(ControlLabelComponent)`
  ${bgColor}
  ${colorForLabel}
  ${getFontFamily}
  ${fontSizeForLabel}
  ${getFontStyle}
  ${getFontWeight}
  ${getLetterSpacing}
  ${getLineHeight}
  ${getTextAlign}
  ${getTextDecoration}
  left: ${padding};
  padding: 0 ${padding};
  ${topStyle};
  ${positionStyle}
`;

ControlLabel.propTypes = {
  bgColor: PropTypes.string,
  disabled: PropTypes.bool,
  isRelative: PropTypes.bool,
  ...typography.propTypes,
  validationError: PropTypes.string,
};

ControlLabel.defaultProps = {
  bgColor: defaultControlStylesBase.bgColor,
  disabled: false,
  isRelative: false,
  validationError: '',
};
