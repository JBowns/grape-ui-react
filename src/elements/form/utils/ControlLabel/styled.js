import PropTypes from 'prop-types';
import styled from 'styled-components';
import { getGlobalStyles } from 'src/global-styles';
import {
  fontWeight,
  layout,
  position,
  space,
} from 'styled-system';
import {
  colorCore,
  defaultStylesBase,
  defaultControlStyles,
  ellipsisCore,
  fontFamilyCore,
  fontSizeCore,
  fontStyleCore,
  letterSpacingCore,
  lineHeightCore,
  textAlignCore,
  textDecorationCore,
  typography,
} from 'src/utils/styledHelpers';
import { ControlLabelComponent } from './component';

const { fontSize: fontSizeSchema } = getGlobalStyles();

const colorLabel = props => colorCore({
  ...props,
  color: !props.validationError ? props.color : 'brandDanger',
});
const fontSizeLabel = props => fontSizeCore({ ...props, sm: true });
const ControlLabel = styled(ControlLabelComponent)`
  ${colorLabel}
  ${ellipsisCore}
  ${fontFamilyCore}
  ${fontSizeLabel}
  ${fontStyleCore}
  ${fontWeight}
  ${layout}
  ${letterSpacingCore}
  ${lineHeightCore}
  ${position}
  ${space}
  ${textAlignCore}
  ${textDecorationCore}
`;

ControlLabel.propTypes = {
  /** Background of the label component.  For 'outlined' styled controls, this assures that the label is making space on the control's border. */
  bg: PropTypes.string,
  /** When true, this will change the label text color to the control disabled color. */
  disabled: PropTypes.bool,
  ...typography.propTypes,
  /** When true, this will change the label text color to brandDanger. */
  validationError: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
  ]),
};

ControlLabel.defaultProps = {
  ...defaultStylesBase,
  bg: defaultControlStyles.bg,
  disabled: false,
  height: `${fontSizeSchema.sizeVariants.sm / 2}rem`,
  left: 2,
  lineHeight: 0.8,
  position: 'absolute',
  px: 2,
  top: '-1px',
  validationError: '',
};

/** @component */
export { ControlLabel };
