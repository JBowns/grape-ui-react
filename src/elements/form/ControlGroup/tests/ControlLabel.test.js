import React from 'react';
import { ControlLabel } from '../ControlLabel';
import 'jest-styled-components';
// import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('ControlLabel Component base', () => {
  it('should return a ControlLabel object', () => {
    const component = mount(<ControlLabel for="forWhatId">Label text is here</ControlLabel>);
    expect(toJson(component.find('ControlLabel'))).toMatchSnapshot();
  });
});
