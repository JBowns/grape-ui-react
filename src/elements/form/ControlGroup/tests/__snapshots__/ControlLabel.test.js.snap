// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlLabel Component base should return a ControlLabel object 1`] = `
.c0 {
  background-color: white;
  left: 0.5rem;
  padding: 0 0.5rem;
  position: absolute;
  top: -0.5rem;
  font-family: inherit;
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<ControlLabel
  bgColor="white"
  color="inherit"
  fontFamily="inherit"
  fontWeight="inherit"
  for="forWhatId"
  kerning="inherit"
  lg={false}
  sm={false}
  textAlign="inherit"
  textDecoration="inherit"
  validationError=""
>
  <styled.label
    for="forWhatId"
  >
    <label
      className="c0"
    >
      Label text is here
    </label>
  </styled.label>
</ControlLabel>
`;
